src/v2i-hub/CARMACloudPlugin/src/CARMACloudPlugin.cpp

```diff
@@ -474,29 +466,31 @@ void CARMACloudPlugin::OnStateChange(IvpPluginState state) {
 
 
 int CARMACloudPlugin::CloudSend(string msg,string url, string base, string method)
-{
-	CURL *req;
-  	CURLcode res;
-	string urlfull = url+base;	  
-
-
-  	req = curl_easy_init();
- 	 if(req) {
-  	  	curl_easy_setopt(req, CURLOPT_URL, urlfull.c_str());
-
-		if(strcmp(method.c_str(),"POST")==0)
-		{
-    		curl_easy_setopt(req, CURLOPT_POSTFIELDS, msg.c_str());
-			curl_easy_setopt(req, CURLOPT_TIMEOUT, 2L); // Sets a 2 second timeout 
-			res = curl_easy_perform(req);
-   			if(res != CURLE_OK)
-			   {
-      				fprintf(stderr, "curl send failed: %s\n",curl_easy_strerror(res));
-					  return 1;
-			   }	  
-		}
-    	curl_easy_cleanup(req);
-  }
+{ 
+	std::thread t([msg, url, base, method](){
+		CURL *req;
+		CURLcode res;
+		string urlfull = url+base;	
+		req = curl_easy_init();
+		if(req) {
+			curl_easy_setopt(req, CURLOPT_URL, urlfull.c_str());
+
+			if(strcmp(method.c_str(),"POST")==0)
+			{
+				curl_easy_setopt(req, CURLOPT_POSTFIELDS, msg.c_str());
+				curl_easy_setopt(req, CURLOPT_TIMEOUT_MS, 1000L); // Request operation complete within max millisecond timeout 
+				res = curl_easy_perform(req);
+				if(res != CURLE_OK)
+				{
+						fprintf(stderr, "curl send failed: %s\n",curl_easy_strerror(res));
+						return 1;

```

Comment:
Does return 1 do anything in a detached thread?

Reply:
It should exit.